src/templates/common/.gitlab-ci.yml

```diff
@@ -1,9 +1,19 @@
+image: julia:1.5
+
+before_script:
+  - apt-get -qq update; apt-get -y install git
+
 pages:
-    stage: deploy
-    script:
-        - mv __site public
-    artifacts:
-        paths:
-            - public
-    only:
-        - master
+  stage: deploy
+  script:
+    - julia --project=@. -e 'import Pkg; Pkg.instantiate();
+      using NodeJS; run(`$(npm_cmd()) install highlight.js`);
+      using Franklin;
+      serve(single = true, is_final_pass = true);
```

Comment:
this line `(serve ...)` is not needed (if you want to convince yourself, have a look at the implementation of `optimize`: https://github.com/tlienart/Franklin.jl/blob/2db712b5d4bcf28dde4b44042e88e6b1dc87f163/src/manager/post_processing.jl#L55-L58 ;-)

Reply:
Didn't look at that function beforehand, seems like it does what's needed. Will remove it.